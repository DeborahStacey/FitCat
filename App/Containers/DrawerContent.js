import React, { Component } from 'react'
import { AsyncStorage, ScrollView } from 'react-native'
import I18n from 'react-native-i18n'
import styles from './Styles/DrawerContentStyle'
import DrawerButton from '../Components/DrawerButton'
import { Actions as NavigationActions } from 'react-native-router-flux'
import { default as StorageKeys } from '../Config/StorageKeys'

class DrawerContent extends Component {

  toggleDrawer () {
    this.context.drawer.toggle()
  }

  handlePressComponents = () => {
    this.toggleDrawer()
    NavigationActions.componentExamples()
  }

  handlePressUsage = () => {
    this.toggleDrawer()
    NavigationActions.usageExamples()
  }

  handlePressAPI = () => {
    this.toggleDrawer()
    NavigationActions.apiTesting()
  }

  handlePressTheme = () => {
    this.toggleDrawer()
    NavigationActions.theme()
  }

  handlePressChangeCat = () => {
    this.toggleDrawer()
    NavigationActions.addCat()
  }

  handlePressLogOut = () => {
    this.toggleDrawer()
    AsyncStorage.setItem(StorageKeys.WELLCAT_EMAIL, '')
    AsyncStorage.setItem(StorageKeys.FITBIT_ACCESS_TOKEN, '')
    NavigationActions.notLoggedInDash()
  }

  handlePressFoodConsumption = () => {
    this.toggleDrawer()
    NavigationActions.foodConsumption()
  }

  render () {
    return (
      <ScrollView style={styles.container}>
        <DrawerButton text={I18n.t('change_cat')} onPress={this.handlePressChangeCat} />
        <DrawerButton text={I18n.t('food_consumption')} onPress={this.handlePressFoodConsumption} />
        <DrawerButton text={I18n.t('logOut')} onPress={this.handlePressLogOut} />
        {/* <DrawerButton text='Component Examples' onPress={this.handlePressComponents} />
        <DrawerButton text='Usage Examples' onPress={this.handlePressUsage} />
        <DrawerButton text='API Testing' onPress={this.handlePressAPI} />
        <DrawerButton text='Themes' onPress={this.handlePressTheme} />
        <DrawerButton text='Device Info' onPress={this.handlePressDevice} /> */}
      </ScrollView>
    )
  }

}

DrawerContent.contextTypes = {
  drawer: React.PropTypes.object
}

export default DrawerContent
