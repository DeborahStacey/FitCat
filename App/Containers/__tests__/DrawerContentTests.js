import React from 'react'
import DrawerContent from '../DrawerContent'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

it('renders the navigation drawer correctly', () => {
  const tree = renderer.create(
    <DrawerContent />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
