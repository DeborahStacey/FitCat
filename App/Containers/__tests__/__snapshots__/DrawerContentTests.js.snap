exports[`test renders the navigation drawer correctly 1`] = `
<ScrollView
  style={
    Object {
      "flex": 1,
      "padding": 20,
    }
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#2C457F",
          "fontFamily": "HelveticaNeue",
          "fontSize": 20,
          "marginVertical": 10,
        }
      }>
      [missing \"en.change_cat\" translation]
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#2C457F",
          "fontFamily": "HelveticaNeue",
          "fontSize": 20,
          "marginVertical": 10,
        }
      }>
      [missing \"en.new_cat\" translation]
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#2C457F",
          "fontFamily": "HelveticaNeue",
          "fontSize": 20,
          "marginVertical": 10,
        }
      }>
      [missing \"en.logOut\" translation]
    </Text>
  </View>
</ScrollView>
`;
